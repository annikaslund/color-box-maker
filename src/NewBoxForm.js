import React, { Component } from "react";
import uuid from "uuid/v4";
import "./NewBoxForm.css";

class NewBoxForm extends Component {
    constructor(props){
        super(props);
        this.state = {
            height: 0,
            width: 0,
            bgColor: ""
        }
        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleSubmit(evt){
        evt.preventDefault();
        this.props.addBox(this.state);
        this.setState({
            height: 0,
            width: 0,
            bgColor: ""
        });
    }
    
    handleChange(evt){
        this.setState({
            [evt.target.name]: evt.target.value
        })
    }

    render() {
        return (
            <form className="NewBoxForm" onSubmit={ this.handleSubmit }>
                <label forHTML="height">Height:</label>
                <input name="height" value={ this.state.height } onChange={ this.handleChange }></input>
                
                <label forHTML="width">Width:</label>
                <input name="width" value={ this.state.width } onChange={ this.handleChange }></input>

                <label forHTML="bgColor">Background Color:</label>
                <input name="bgColor" value={ this.state.bgColor } onChange={ this.handleChange }></input>
            </form>
        );
    }
}
export default NewBoxForm;