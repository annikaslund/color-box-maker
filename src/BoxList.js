import React, { Component } from "react";
import uuid from "uuid/v4";
import "./BoxList.css";
import NewBoxForm from './NewBoxForm';
import Box from './Box';

class BoxList extends Component {
    constructor(props) {
        super(props);
        this.state = {
            boxes: []
        };
        this.createNewBox = this.createNewBox.bind(this);
    }

    createNewBox(box) {
        let newBox = { ...box, id: uuid() };
        this.setState(state => ({
            boxes: [...state.boxes, newBox]
        }));
    }

    render() {
        const boxes = this.state.boxes.map(b => (
            <Box key={b.id} height={b.height} width={b.width} bgColor={b.bgColor} />
        ))

        return (
            <div>
                <NewBoxForm getNewBoxInfo={this.createNewBox} />
                {boxes}
            </div>
        )
    }
}

export default BoxList;