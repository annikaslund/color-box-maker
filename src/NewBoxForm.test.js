import React from "react";
import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import BoxList from "./BoxList";
import NewBoxForm from "./NewBoxForm";

it("renders without crashing", function(){
    mount(<NewBoxForm />);
})

it("matches snapshot", function(){
    let wrapper = mount(<NewBoxForm />);
    let serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
})