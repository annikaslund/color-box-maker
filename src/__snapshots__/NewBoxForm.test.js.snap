// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<NewBoxForm>
  <form
    className="NewBoxForm"
    onSubmit={[Function]}
  >
    <label
      htmlFor="height"
    >
      Height:
    </label>
    <input
      name="height"
      onChange={[Function]}
      value={0}
    />
    <label
      htmlFor="width"
    >
      Width:
    </label>
    <input
      name="width"
      onChange={[Function]}
      value={0}
    />
    <label
      htmlFor="bgColor"
    >
      Background Color:
    </label>
    <input
      name="bgColor"
      onChange={[Function]}
      value=""
    />
    <button>
      Make Button!
    </button>
  </form>
</NewBoxForm>
`;
